lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetSchemaConverter.java

```diff
@@ -148,6 +151,10 @@ private org.apache.parquet.schema.Type getPrimitiveType(Type type, String name,
 
         if (type instanceof TimestampType) {
             TimestampType timestampType = (TimestampType) type;
+            // Apache Hive version 3 or lower does not support reading timestamps encoded as INT32/INT64
```

Comment:
What about newer versions of Hive, or Spark?

Reply:
Hive 4 started supporting int64 backed timestamps https://issues.apache.org/jira/browse/HIVE-21216
Spark has supported int64 timestamps since some 2.x version

Reply:
Some specific Hive distros also cherry-picked support for INT64 timestamps into earlier version. Cloudera's CDP7  is based on Hive 3 but has support for using INT64